lib/users.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const isFunction = require('101/is-function')
+const Promise = require('bluebird')
+const Util = require('./util')
+
+/**
+ * Users - Used for communication with intercomClient's user object
+ * @class
+ * @author Ryan Kahn
+ */
+class Users {
+  /**
+   * Constructor for Users
+   * @param {object} client Intercom client for customer related actions.
+   */
+  constructor (client) {
+    this.client = client
+  }
+
+  /**
+   * Helper method to wrap all the intercomClient.users calls around
+   * `canUseIntercom`
+   * @param {String} method Name of the method to wrap for the intercom
+   *   users client.
+   * @returns {Promise} - Returns success if intercom is disabled, otherwise
+   *   returns results of call to intercom client
+   * @private
+   */
+  _wrap (method) {
```

Comment:
If we had another library I think we should abstract this out into a base class. But for now I think it's alright.